scripts/inner/kubestellar

```diff
@@ -227,23 +199,27 @@ function run_dex() {
     fi
 }
 
-function run_kb_backend() {
+# ensure_kb_backend takes one positional parameter:
+# - the file pathname of the kubeconfig to use for access to the KCS,
+#   the caller may modify or delete this file after this function returns.
+function ensure_kb_backend() {
     if [ $(process_running example-backend) == "running" ]; then
         echo "example-backend is already running"
         return
     fi
 
     echo "starting kube-bind example-backend in background"
 
-    kubectl-kubestellar-get-config-for-space --space-name "espw" --sm-core-config $SM_CONFIG ${in_cluster} --sm-context $SM_CONTEXT --output "/tmp/espw-conf"
+    local kcs_kubeconfig="/tmp/kcs.kubeconfig"
+    cp "$1" $kcs_kubeconfig
     
     ( 
-        KUBECONFIG="/tmp/espw-conf"
+        export KUBECONFIG="$kcs_kubeconfig"
```

Comment:
Is the scope of this KUBECONFIG only within the round brackets?  Or does this change the value of KUBECONFIG outside?

Reply:
Round brackets make a subshell. A subprocess does not affect the environment of its ancestors or siblings.